README.md

```diff
@@ -1,42 +1,99 @@
-Dynamic Extensions for Alfresco
-===============================
+# Dynamic Extensions for Alfresco
 
 [![Build Status](https://travis-ci.org/xenit-eu/dynamic-extensions-for-alfresco.svg)](https://travis-ci.org/xenit-eu/dynamic-extensions-for-alfresco)
 
+Add OSGi based hot-deploy functionality and Spring annotation based configuration to your Alfresco.
+
+## Introduction
+
 Rapid development of Alfresco repository extensions in Java. Deploy your code in seconds, not minutes. Life is too short for endless server restarts.
 
 Dynamic Extensions adds an OSGi container to the Alfresco repository, enabling live deployment of Java code, with no need to restart the server.
 Alfresco itself is not "OSGi-fied" in any way; the OSGi container runs on top of the core Alfresco platform.
 
-Installing Dynamic Extensions
------------------------------
+Default Alfresco Platform Extensions use Spring XML based configuration. With Dynamic Extensions, developers
+have the ability to create Alfresco Platform Extensions using Spring's annotations based configuration.
 
-To make sure Dynamic Extensions is build and compiled against the correct
-Alfresco version, build artifacts for each minor Alfresco version update 
-are distributed. E.g. if you are working with Alfresco 6.0.7-ga, you should
-use the `alfresco-dynamic-extensions-repo-60` artifact.
+## Example Dynamic Extensions based Alfresco Platform extension
 
-Dynamic Extensions is distributed as an Alfresco Module Package (AMP).
+This example Web Script examines a node and passes information to a Freemarker template:
+```java
+@Component
+@WebScript
+public ExampleWebScript {
+
+  @Autowired
+  private NodeService nodeService;
 
-### Maven Central
+  @Uri("/show-node")
+  // Example: http://localhost/alfresco/service/show-node?nodeRef=workspace://SpacesStore/12345
+  public Map<String, Object> displayNodeName(@RequestParam NodeRef nodeRef) {
+    Map<String, Object> model = new HashMap<String, Object>();
+    model.put("properties", nodeService.getProperties(nodeRef));    
+    return model; // Model is passed to Freemarker template.
+  }
+}
 ```
+
+Note that this is an _annotation Web Script_. These types of Web Script are configured through Java annotations instead of `*.desc.xml` descriptors. Annotation Web Scripts are similar to Spring MVC's annotation-based controllers.
+
+Here's the accompanying Freemarker template fragment:
+
+```html

```

Comment:
I think it needs to be mentionned where this file has to be placed.

Reply:
This is mentioned in the other documentation, KIS here?

Reply:
Okay